octodns/provider/base.py

```diff
@@ -28,26 +30,19 @@ def __init__(self, id, apply_disabled=False,
         self.apply_disabled = apply_disabled
         self.update_pcent_threshold = update_pcent_threshold
         self.delete_pcent_threshold = delete_pcent_threshold
-
-    def _include_change(self, change):
-        '''
-        An opportunity for providers to filter out false positives due to
-        peculiarities in their implementation. E.g. minimum TTLs.
-        '''
-        return True
-
-    def _extra_changes(self, existing, desired, changes):
-        '''
-        An opportunity for providers to add extra changes to the plan that are
-        necessary to update ancillary record data or configure the zone. E.g.
-        base NS records.
-        '''
-        return []
+        self.strict_supports = strict_supports
```

Comment:
Sounds like it implies that we should check for all `SUPPORTS_*` features in `_process_desired_zone` and not just PTR one? May be good to consolidate all of them in one method.

Reply:
Yeah. That's what I was meaning when I said the larger work here would be sizeable. There'd likely even be some simplification/refactoring if we remove `dynamic` (and `geo`) from the desired during this point of things. 

That's probably too big of a change for a single PR so my thinking would be to ship this once we're happy with it and iterate towards other things happening here that are already happening elsewhere over time/PRs.